ibis-server/app/routers/v3/connector.py

```diff
@@ -88,61 +93,67 @@ async def query(
                     data_source, dto.sql, dto.connection_info
                 )
                 cache_hit = cached_result is not None
+            # case 1: cache hit read
+            # We also need to check `is_fallback_disable` because a previous request
+            # might have fallen back successfully  and created a cache.
+            # If the current request has fallback disabled, we should bypass the cache hit check.
```

Comment:
I think it's wrong. If we always bypass the cache hit check, it means the cache never works for the request, which disables the fallback. IMO, it's fine to use the cache created by the previous fallback request.